profiling/DerSPMF_test.jl

```diff
@@ -0,0 +1,70 @@
+using NonlinearEigenproblems, Random, SparseArrays, Revise, LinearAlgebra, BenchmarkTools
+import ..NEPTypes.AbstractSPMF
+struct DerSPMF{T<:AbstractMatrix,FDtype} <: AbstractSPMF{T}
+    spmf::SPMF_NEP{T}
+    fD::Matrix{FDtype}
+    σ::Number
+end
+
+## one constructor takes spmf as input and compute the derivatives
+function DerSPMF(spmf::SPMF_NEP,σ::Number,m::Int)
+      # Compute DD-matrix from get_fv(spmf)
+      TT=promote_type(typeof(σ),eltype(nep.A[1]))
+      p=length(nep.fi)
+      fD=zeros(TT,m,p)
+      # matrix for the computation of derivatives
+      SS=diagm(0 => σ*ones(TT,2m+2)) + diagm(-1 => (1:2m+1))
+      fD=zeros(2*m+2,p)
```

Comment:
I think this should have a type, like on line 14. And line 14 could be removed, right?